File: java/LinkedList.java
public class LinkedList {
	private class Node {
    	int x;
		Node next;
	}

	private Node head, tail;
	
	LinkedList(Node node) {
		if (node == null)
			throw new Exception("Can't be null");
		this.head = node;
		this.tail = node;
	}

	void addNodeToEnd() {
			
	}
}

File: library/linkedlist.cpp
#include<iostream>
using namespace std;

struct Node {
   int code;
   Node * next;
};

class LinkedList {
    private:
	Node * head;
	Node * tail;
    
    public:
    LinkedList(Node * head) {
		if (head == null)
			throw "HEAD can't be null";
		this->head = head;
		this->tail = head;
	}

	void addNodeToEnd(Node *node) {
		this->tail->next = node;
		node->next = NULL;
		this->tail = node;	
	}

}
